### DISCUSSION TITLE
Help building a standalone project using zipline

### DISCUSSION THREAD

**Discussion by tsouza-squid** [cashapp/zipline#914] [Q&A]
Hello zipline folks,

Currently, we are in the process of developing a cloud service that allows developers to build applications on top of it using a client library that we provide. The library is currently written in TypeScript and contains a significant amount of business logic. Our objective is to enable our users to develop native Android applications in Kotlin that have access to this TypeScript client.

To achieve this goal, we believe that Zipline can be utilized to bundle the TypeScript client in JS and create an adapter in Kotlin/JS. This would enable us to seamlessly integrate the TypeScript client into an Android application.

We are currently in the process of creating a standalone Android library that utilizes Zipline for this purpose. However, we have encountered a challenge with the project's sample structure, which appears to be tightly coupled with the root project. Therefore, we would appreciate any assistance in understanding the ideal directory structure, gradle.settings, and gradle.build for a standalone project that aligns with our objectives.

Thanks

**Comment by swankjesse** [comment]
Apologies for sitting on this for 2 months! Here’s [one sample app](https://github.com/matt-ramotar/pokedex) that uses Zipline and Redwood. You might find other sample projects searching GitHub for [app.cash.zipline](https://github.com/search?q=app.cash.zipline&type=code).
